sdk/spring/spring-cloud-azure-starter-monitor/src/main/java/com/azure/spring/cloud/autoconfigure/monitor/implementation/selfdiagnostics/SpringEnvSelfDiag.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.cloud.autoconfigure.monitor.implementation.selfdiagnostics;
+
+import org.slf4j.Logger;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.Environment;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+class SpringEnvSelfDiag implements CommandLineRunner {
```

Comment:
Replaced execution env self-diagnostics.

These new diagnostics also include data from the Spring files (.properties, .yaml)